### PR TITLE
Fix cloudwatch multivalue

### PR DESCRIPTION
**Pull request by mkm85**
Fixed a bug cloudwatch datasource.js and added handling in the case where All templating options is used.

I'm not 100% sure that the handling of the all option is correct.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/datasource.js [hunk 1]
```diff
@@ -379,7 +389,7 @@ function (angular, _, moment, dateMath, kbn, CloudWatchAnnotationQuery) {
 
         if (dimensionKey) {
           var variable = _.find(templateSrv.variables, function(variable) {
-            return templateSrv.containsVariable(target.dimensions[dimensionKey], variable.name);
+            return self.containsVariable(target.dimensions[dimensionKey], variable.name);
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: why this change?

Comment (mkm85) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: templateSrv does not contain the function containsVariable any more. Which means the current master does not work with muliple selected values.

File: public/app/plugins/datasource/cloudwatch/datasource.js [hunk 2]
```diff
@@ -358,9 +358,15 @@ function (angular, _, moment, dateMath, kbn, CloudWatchAnnotationQuery) {
     }
 
     this.getExpandedVariables = function(target, dimensionKey, variable) {
+      /* if the all checkbox is marked we should add all values to the targets */
+      var allSelected = _.find(variable.options, {'selected': true, 'text': 'All'});
```
Comment (mkm85) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: When selecting all, the variable.current.value contains $__all if that can easily be expanded to all the values the code can be made better.

File: public/app/plugins/datasource/cloudwatch/datasource.js (hunk 1 above)
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: that is true, it's here https://github.com/grafana/grafana/blob/46d4f817e3f6e0e7ca39027b155d084d14dd2d8a/public/app/features/templating/variable.ts#L23 

But why the contains check? templating is for interpolation. But for cloudwatch it is not used for interpolation, but looking up the variable and it's values, so why not just look up the variable by name check? no need to use contains

File: public/app/plugins/datasource/cloudwatch/datasource.js (hunk 2 above)
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: you will have to modify templateSrv it does that.

File: public/app/plugins/datasource/cloudwatch/datasource.js (hunk 1 above)
Comment (mkm85) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: Short answer, the code already did contain a contains check. I'll see if I have time to look into it later this week. You should feel free to just create a fix such that it will be working on the master again.

**Comment by jangaraj** [comment]
Could you provide some example how to use cloudwatch multivalue please? Maybe doc http://docs.grafana.org/datasources/cloudwatch/ should be updated as well. Thx.
